site/pages/view-all-disruptions.page.tsx

```diff
@@ -105,25 +106,33 @@ const getDisruptionStatus = (disruption: SortedDisruption): string => {
     }
 
     if (!disruption.validity) {
-        return "closed";
+        return Progress.closed;
     }
 
     const today = getDate();
     const disruptionEndDate = getSortedDisruptionFinalEndDate(disruption);
 
     if (!!disruptionEndDate && dateIsSameOrBeforeSecondDate(disruptionEndDate, today)) {
-        if (disruptionEndDate.isBefore(today)) {
-            return "closed";
-        } else {
-            return "closing";
-        }
+        return disruptionIsClosingOrClosed(disruptionEndDate, today);
     }
 
-    return "open";
+    return Progress.open;
 };
 
-const formatDisruptionsIntoRows = (disruptions: TableDisruption[], offset: number) => {
-    return disruptions.map((disruption, index) => {
+export const disruptionIsClosingOrClosed = (disruptionEndDate: Dayjs, today: Dayjs): Progress => {
+    if (disruptionEndDate.isBefore(today)) {
+        return Progress.closed;
+    } else if (disruptionEndDate.diff(today, "hour") < 24) {
+        return Progress.closing;
```

Comment:
Closing status not displayed for a new or edited disruption when the end date is set to next day with end time set to 22 hours from now